### ISSUE TITLE
Restriction on the number of equality constraints

### ISSUE THREAD

**Issue by tkx68** [stevengj/nlopt#357]
The NLopt optimizer returns NLOPT_INVALID_ARGS if you give more than n equality constraints in a problem with a variables. This is totally reasonable in linear settings but not necessarily in nonlinear problems. As far as I can see e.g. ISRES does not depend on this. Could we lift this restriction and maybe just offer a warning with the result?

This feature/restriction was introduced in 2010 as can be seen in the NEWS file.

The problem arises in the functions `nlopt_add_equality_mconstraint` and `nlopt_add_precond_equality_constraint` in the file `options.c` where `nlopt_count_constraints` is called.

**Comment by stevengj** [comment]
Yes, I think it is fine to remove this check and just expect people to know what they are doing.

**Comment by m-zanichelli** [comment]
Hi, any updates on this?
